web/modules/custom/product_uploader/src/ProductUploaderInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\product_uploader;
+
+/**
+ * Class for the Product Uploder role.
+ */
+class ProductUploaderInterface {
```

Comment:
Is this a class or an interface?

I think this should be a `final class` with an empty private constructor: https://stackoverflow.com/a/53194084